dockeradapter/dockeradapter.go

```diff
@@ -130,18 +141,91 @@ func ImageExists(ctx context.Context, client ImageInspector, imageName string) b
 	return true
 }
 
-// New returns a client satisfying the Client interface
+// New returns a client connected with the highest API version supported by both the Lumogon client
+// and the Docker runtime
 func New() (Client, error) {
-	concreteClient := new(concreteDockerClient)
-	logging.Debug("[Docker Adapter] Creating container runtime client: Docker")
-	dockerAPIClient, err := client.NewEnvClient()
+
+	host, _, _ := DockerEnvvars()
+
+	serverAPIVersion, _, err := ServerInfo(host)
 	if err != nil {
-		return nil, fmt.Errorf("[Docker Adapter] Unable to initialise container runtime type: Docker, error: %s", err)
+		logging.Debug("[Docker Adapter] Unable to determine Docker Server API version: %s", err.Error())
+		return nil, err
 	}
-	concreteClient.Client = dockerAPIClient
+
+	// Connect using the API version that the server supports
+	client, err := client.NewClient(host, serverAPIVersion, nil, nil)
+	if err != nil {
+		logging.Debug("[Docker Adapter] Unable to connect to Docker server")
+		return nil, err
+	}
+
+	logging.Debug("[Docker Adapter] Creating container runtime client: Docker")
+	concreteClient := new(concreteDockerClient)
+	concreteClient.APIVersion = serverAPIVersion
+	concreteClient.Client = client
+
 	return concreteClient, nil
 }
 
+// DockerEnvvars returns values for the following environment variables, setting a default
+// if no variable is set: DOCKER_HOST, DOCKER_CERT_PATH, DOCKER_TLS_VERIFY
+func DockerEnvvars() (string, string, bool) {
```

Comment:
I feel this is misnamed, based on the implementation. The defaults means this is much more than just the envvars. I'd change this to DockerConfig or similar?